src/numerics/petsc_matrix.C

```diff
@@ -470,6 +470,14 @@ void PetscMatrix<T>::update_preallocation_and_zero ()
   libmesh_not_implemented();
 }
 
+template <typename T>
+void PetscMatrix<T>::reset_preallocation()
+{
+  libmesh_assert (this->initialized());
+
+  auto ierr = MatResetPreallocation(_mat);
```

Comment:
Version control

Reply:
Ugh @fdkong you told me the wrong version number [here](https://github.com/idaholab/moose/pull/13670#discussion_r299159181). Hence the test failures on idaholab/moose#13682. Looks like it should be 3.9.0 based on `git tag --contains`. I'm guessing you just made a typo sigh.

Reply:
https://www.mcs.anl.gov/petsc/petsc-3.9/docs/manualpages/Mat/MatResetPreallocation.html

It is in 3.9. I am confused here.

Reply:
`!PETSC_VERSION_LESS_THAN(3,8,0)` means >= 3.8.0, i.e. including 3.8.0

Reply:
Right.

Reply:
We need >=3.9.0